### ISSUE TITLE
OpenShot v.2.4.3 crashes during export process

### ISSUE THREAD

**Issue by andresaluki** [OpenShot/openshot-qt#2228]
Hi there!

**Describe the bug**
I can't finish exporting my project. The launch.exe is crashing during the process.

**System Details (please complete the following information):**
 - Operating System / Distro: Windows 7 Ultimate 32 bits
 - OpenShot Version 2.4.3

**To Reproduce**
1. Go to 'File - Recent Projects - my project'
2. Click on 'Export video - Video profile - HD 1080 30 fps (1920x1080) - Export video'
3. See error: a Windows popup appears: the launch.exe has stopped working.

**Expected behavior**
It was expected the export process to finish normally, even despite the long duration of it due to modest computer configurations.

**Logs**

Update: I manage to find the openshot-qt text file from the openshot-qt folder. The file is new and was created for the last export attempt. I attached it in this message.
[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/2475888/openshot-qt.log)

This is the information showed in the popup error message:

launch.exe parou de funcionar

Assinatura do problema:
  Nome do Evento de Problema:	APPCRASH
  Nome do Aplicativo:	launch.exe
  Versão do Aplicativo:	0.0.0.0
  Carimbo de Data/Hora do Aplicativo:	00000000
  Nome do Módulo de Falhas:	libopenshot.dll
  Versão do Módulo de Falhas:	0.0.0.0
  Carimbo de Data/Hora do Módulo de Falhas:	5ba69d8b
  Código de Exceção:	c0000094
  Deslocamento de Exceção:	00072cec
  Versão do sistema operacional:	6.1.7600.2.0.0.256.1
  Identificação da Localidade:	1046
  Informações Adicionais 1:	0a9e
  Informações Adicionais 2:	0a9e372d3b4ad19135b953a78882e789
  Informações Adicionais 3:	0a9e
  Informações Adicionais 4:	0a9e372d3b4ad19135b953a78882e789

**Additional context**
No other program (web browser, image/audio editing tools, audio/video player) is running in parallel to OpenShot. The moment when the crash happens varies; it already happened on 50%, 60% and 70% of export process. I've already increased the cache size at the maximum possible and unchecked the "process video frames in parellel" option, but the issue persists. I need to export in 1080p 30fps to YouTube as it is the project profile. Reducing the video/audio bitrate quality did not solve the issue. There are plenty of RAM memory available (not being used by OpenShot) during the export process (around 35-40%).

**Comment by cncook001** [comment]
I have a similar crash.

Windows 10
- OpenShot 2.4.3 production release fails, so does the daily
- OpenShot-v2.4.3-dev1-1544831681-a8f75a5b-3861a2c5-x86_64

[openshot-qt.log](https://github.com/OpenShot/openshot-qt/files/2683588/openshot-qt.log)

**Comment by SawyerUSC** [comment]
Has the exporting issue been fixed yet? OpenShot crashes at 76% when exporting a video. I've tried downloading an older version of OpenShot and I've tried exporting in different formats.

**Comment by andresaluki** [comment]
Hi! Thanks for messaging me.

I was using the 32 bits version of Windows 7 Ultimate when this issue was happening. After have made a clean installation of the 64 bits version of Windows 7 Ultimate the issue has stopped happening. I didn't noticed any improvements in performance though, but this is due to the modest hardware settings of my computer. So I guess this issue only occurs in 32 bits OS versions.

Best regards,

Andre

Em 24 de jan de 2019 14:03, SawyerUSC <notifications@github.com> escreveu:

Has the exporting issue been fixed yet? OpenShot crashes at 76% when exporting a video. I've tried downloading an older version of OpenShot and I've tried exporting in different formats.

**Comment by SawyerUSC** [comment]
@andresaluki Hi Andre, thanks for the response. I am using Windows 10 Home with a 64-bit operating system and I've downloaded the 64-bit version of OpenShot, but the problem still persists for me. Maybe there's another issue that's causing it for me. I'm happy to hear that yours is working now.

**Comment by andresaluki** [comment]
Updating the issue:

I thought that the issue had stopped happening after changing my OS from 32 to 64 bits since I had been able to export a short video (about 1 minute duration) without problems. However, when I tried to export a longer video (about 10 minutes duration), the issue happened again, with only 32 - 37% of progress, twice. For me, one thing was very apparent: _memory leaking_. The memory usage by the program always increases with time, whether during the edit process or during the export process. Whenever the percentage of total memory usage reaches over 90%, the issue is about to happen, and it happens.

This is very frustrating. Yesterday I lost an entire day (and night) of working, and I'm exausted. I'll have to give a break from this program until this issue be finally fixed.

Best regards.
